packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -100,39 +120,59 @@ export default function DataViews< Item >( {
 			onChangeSelection( newValue );
 		}
 	}
+
 	const _fields = useMemo( () => normalizeFields( fields ), [ fields ] );
 	const _selection = useMemo( () => {
-		return selection.filter( ( id ) =>
-			data.some( ( item ) => getItemId( item ) === id )
+		return (
+			selection?.filter(
+				( id ) => data?.some( ( item ) => getItemId( item ) === id )
+			) || []
 		);
 	}, [ selection, data, getItemId ] );
 
-	const filters = useFilters( _fields, view );
+	const filters = useFilters( _fields, view ?? ( {} as View ) );
 	const [ isShowingFilter, setIsShowingFilter ] = useState< boolean >( () =>
 		( filters || [] ).some( ( filter ) => filter.isPrimary )
 	);
 
+	const contextValue: DataViewsContextType< Item > = {
+		view: view as View,
+		onChangeView: onChangeView as ( view: View ) => void,
+		fields: _fields,
+		actions,
+		data: ( data || [] ) as Item[],
+		isLoading,
+		paginationInfo: paginationInfo || {
+			totalItems: 0,
+			totalPages: 0,
+		},
+		selection: _selection,
+		onChangeSelection: setSelectionWithChange,
+		openedFilter,
+		setOpenedFilter,
+		getItemId: getItemId as ( item: Item ) => string,
+		getItemLevel,
+		isItemClickable,
+		onClickItem,
+		containerWidth,
+	};
+
+	if ( children ) {
```

Comment:
We can avoid the condition entirely by just putting the fallback DataViews UI in a `DefaultUI ` component and doing `children = <DefaultUI />` or something.